src/components/Pages/Auth/Registration/FormCompletion.js

```diff
@@ -65,16 +67,23 @@ export default function FormCompletion({
     setUserName(e.target.value);
     onUsernameChange(e.target.value);
     setUserNameValid(false);
+
+    console.log("USERNAME_VALID IS NOW FALSE: ", userNameValid);
   }
 
   const suggestUsername = async () => {
-    const template = firstName.charAt(0).toUpperCase() + firstName.substring(1) + lastName.charAt(0).toUpperCase() + lastName.substring(1);
+    let last = "";
+    for (let x of lastName.split('-')) {
+        last += x.charAt(0).toUpperCase()
+    }
+
+    const template = firstName.charAt(0).toUpperCase() + firstName.substring(1) + last;
     const data = await validateUserName(template);
     console.log("data: ", data);
 
-    data[0] ? setUserName(template) : setUserName(data[1]);
+    setUserName(data['suggested_username'])
```

Comment:
need to deal with the case where validateUserName returns false (api call error).